src/tooling/tooling-simple-types.md

```diff
@@ -0,0 +1,232 @@
+    // Here you can define new functions specific to this class
+    // #[signal] is not valid for Resource types.
+    // #[func] makes a function exposed to Godot
+    // ^^ The fn signature must be compatible types!
+
+    // Using &mut self is typical for functions exposed to Godot.
+    // It isn't strictly necessary, but many different calls
+    // require mutability.
+    #[func]
+    fn foo(&mut self) {
+        godot_print!("bar");
+    } 
+}
+
+```
+
+
+### Node Type
+
+Node types are custom nodes which can be added just like any other built-in node (through the "Add Node" menu).
+
+
+```rust
+
+#[derive(GodotClass)]
+#[class(base=Node)]
+struct NodeType {
+
+    // custom values added here with 
+    // #[var] to be a public variable and 
+    // #[export] to export
+
+    #[base]
+    base : Base<Node>
+}
+
+#[godot_api]
+impl INode for NodeType {
+    fn init(base : Base<Node>) -> Self {
+        Self {
+            base,
+        }
+    }
+
+    // here you can override inherited Object functions. 
+    // refer to the Godot documentation for more details on these functions
+}
+
+#[godot_api]
+impl ObjectType {
+    // Here you can define new functions specific to this class
+    // #[signal] is not valid for Object types.
+    // #[func] makes a function exposed to Godot
+    // ^^ The fn signature must be compatible types!
+
+    // signal functions should be left empty. 
+    // connecting the signal name is the same as the
+    // function name as it is shown here.
+    #[signal]
+    fn do_bar() {}
+
+    // Using &mut self is typical for functions exposed to Godot.
+    // It isn't strictly necessary, but many different calls
+    // require mutability.
+    #[func]
+    fn foo(&mut self) {
+        godot_print!("bar");
+    } 
+}
+
+```
+
+
+## Custom Icons
+
+By default, all of your custom types will use the `Node` icon. While this can be serviceable, you will likely want to add custom icons, especially
+if you plan to distribute this to others!
+
+All of the icons must be registered by Type Name in your `.gdextension` file.
+
+```TOML
+
+;; This is a comment
+
+[icons]
+
+TypeName = "res://addons/your_extension/file_name.svg"
+
+```
+
+```admonish hint
+Note that the path is based off of the `res://` path, like all other resources for Godot. This means that you will need to have a standard 
+installation path for your tool. It is recommended to make this in the "addons" folder because this will clearly signify to users that it is a 
+directory of third party resources and code and that they likely shouldn't mess with it unless they are certain of what they are doing.
+```
+
+
+### Formatting for custom icons
+
+The Godot docs do have a [page dedicated](https://docs.godotengine.org/en/stable/contributing/development/editor/creating_icons.html) to tools and
+resources for creating custom icons. The long and short of it is:
+
+- make it an SVG
+- make it 16 pixels by 16 pixels (in the document details of your SVG editor)
```

Comment:
An SVG doesn't have pixels...? 😕

Reply:
From the Godot documentation they suggest the internal document be labeled as 16x16 pixels and doing otherwise will actually create an oddly scaled image in-editor IIRC (It's been a while since I didn't set the size). Overall, I kinda dislike the icons system Godot has, but I wanted to make it extra clear that it's an important step

Reply:
[Godot's doc](https://docs.godotengine.org/en/stable/contributing/development/editor/creating_icons.html#creating-icons) mentions:

> Icons must be 16×16. In Inkscape, you can configure the document size in File > Document Properties.

These are not pixels though (an SVG is vector based), I think it's just a way to standardize the size and aspect ratio.